junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/OrderedClassTests.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.engine.Constants.CLASS_ORDERING_STRATEGY_PROPERTY_NAME;
+import static org.junit.jupiter.engine.Constants.CUSTOM_CLASS_ORDERER_PROPERTY_NAME;
+import static org.junit.jupiter.engine.Constants.DEFAULT_TEST_METHOD_ORDER_PROPERTY_NAME;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.ClassOrderer;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.platform.testkit.engine.EngineTestKit;
+import org.junit.platform.testkit.engine.Events;
+
+class OrderedClassTests {
+
+	private static final Set<String> callSequence = Collections.synchronizedSet(new LinkedHashSet<>());
+
+	@BeforeEach
+	void clearCallSequence() {
+		callSequence.clear();
+	}
+
+	@Test
+	void className() {
+		var tests = executeTests(ClassOrderer.ClassName.class);
+
+		tests.assertStatistics(stats -> stats.succeeded(callSequence.size()));
+
+		assertThat(callSequence)//
+				.containsExactly("A_TestCase", "B_TestCase", "C_TestCase");
+	}
+
+	@Test
+	void displayName() {
+		var tests = executeTests(ClassOrderer.DisplayName.class);
+
+		tests.assertStatistics(stats -> stats.succeeded(callSequence.size()));
+
+		assertThat(callSequence)//
+				.containsExactly("C_TestCase", "B_TestCase", "A_TestCase");
+	}
+
+	@Test
+	void orderAnnotation() {
+		var tests = executeTests(ClassOrderer.OrderAnnotation.class);
+
+		tests.assertStatistics(stats -> stats.succeeded(callSequence.size()));
+
+		assertThat(callSequence)//
+				.containsExactly("A_TestCase", "C_TestCase", "B_TestCase");
+	}
+
+	@Test
+	void random() {
+		var tests = executeTests(ClassOrderer.Random.class);
+
+		tests.assertStatistics(stats -> stats.succeeded(callSequence.size()));
+	}
+
+	private Events executeTests(Class<? extends ClassOrderer> classOrderer) {
+		// @formatter:off
+        return EngineTestKit
+                .engine("junit-jupiter")
+                .configurationParameter(CLASS_ORDERING_STRATEGY_PROPERTY_NAME, "custom")
+                .configurationParameter(CUSTOM_CLASS_ORDERER_PROPERTY_NAME, classOrderer.getName())
+                .configurationParameter(DEFAULT_TEST_METHOD_ORDER_PROPERTY_NAME, MethodOrderer.MethodName.class.getName())
```

Comment:
I don't think we need to order methods here, do we?

Reply:
Initially I did having multiple test methods, but now that only one is kept it is indeed unnecessary.